File: code/functions/pyprism/pyprism/util/_setwd.py
# -*- coding: utf-8 -*-
"""
@modified: 06 Nov 2020
@created: 30 Sep 2020
@author: Yoann Pradat

    CentraleSupelec
    MICS laboratory
    9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France

    Institut Gustave Roussy
    Prism Center
    114 rue Edouard Vaillant, Villejuif, 94800 France

Fonctions for setting the current working directory to folders of the project.

The following functions are implemented
    setwd_to_data()
    setwd_to_results()
    setwd_to_scripts()
"""

import os

# Name of the project root folder. Modify if you've cloned the git repository under a different name
PROJECT_FOLDER = "MetaPRISM"

def setwd_to_data():
    current_wd = os.getcwd()
    if PROJECT_FOLDER not in os.getcwd():
        raise ValueError("Please set the working directory to a location in the repository %s" % PROJECT_FOLDER)
    else:
        while not os.getcwd().endswith(PROJECT_FOLDER):
            os.chdir("..")
        os.chdir("./data/")
    return current_wd

def setwd_to_results():
    current_wd = os.getcwd()
    if PROJECT_FOLDER not in os.getcwd():
        raise ValueError("Please set the working directory to a location in the repository %s" % PROJECT_FOLDER)
    else:
        while not os.getcwd().endswith(PROJECT_FOLDER):
            os.chdir("..")
        os.chdir("./results/")
    return current_wd


def setwd_to_scripts():
    current_wd = os.getcwd()
    if PROJECT_FOLDER not in os.getcwd():
        raise ValueError("please set the working directory to a location in the repository %s" % PROJECT_FOLDER)
    else:
        while not os.getcwd().endswith(PROJECT_FOLDER):
            os.chdir("..")
        os.chdir("./scripts/")
    return current_wd

File: code/functions/rprism/R/setwd.R
# modified: 19 May 2021
# @created: 06 Nov 2020
# @author: Yoann Pradat
# 
#     CentraleSupelec
#     MICS laboratory
#     9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France
# 
#     Institut Gustave Roussy
#     Prism Center
#     114 rue Edouard Vaillant, Villejuif, 94800 France
# 
# Fonctions for setting the current working directory to folders of the project_
# 
# The following functions are implemented
#     setwd_to_data()
#     setwd_to_results()
#     setwd_to_logs()
#     setwd_to_scripts()

# Name of the project root folder_ Modify if you've cloned the git repository under a different name
PROJECT_FOLDER <- "MetaPRISM"

#' Set the working directory to the data folder of the project.
#'
#' @return current_wd name of the working directory before change
#' @author Yoann Pradat
#' @export
setwd_to_data <- function(){
  current_wd <- getwd()
  if (!grepl(PROJECT_FOLDER, getwd())){
    stop(paste("Please set the working directory to a location in the repository", PROJECT_FOLDER))
  } else {
    while (!endsWith(getwd(), PROJECT_FOLDER)){
      setwd("..")
    }
  }
  setwd("./data")
  return(current_wd)
}

#' Set the working directory to the results folder of the project.
#'
#' @return current_wd name of the working directory before change
#' @author Yoann Pradat
#' @export
setwd_to_results <- function(){
  current_wd <- getwd()
  if (!grepl(PROJECT_FOLDER, getwd())){
    stop(paste("Please set the working directory to a location in the repository", PROJECT_FOLDER))
  } else {
    while (!endsWith(getwd(), PROJECT_FOLDER)){
      setwd("..")
    }
  }
  setwd("./results")
  return(current_wd)
}

#' Set the working directory to the logs folder of the project.
#'
#' @return current_wd name of the working directory before change
#' @author Yoann Pradat
#' @export
setwd_to_logs <- function(){
  current_wd <- getwd()
  if (!grepl(PROJECT_FOLDER, getwd())){
    stop(paste("Please set the working directory to a location in the repository", PROJECT_FOLDER))
  } else {
    while (!endsWith(getwd(), PROJECT_FOLDER)){
      setwd("..")
    }
  }
  setwd("./logs")
  return(current_wd)
}

#' Set the working directory to the scripts folder of the project.
#'
#' @return current_wd name of the working directory before change
#' @author Yoann Pradat
#' @export
setwd_to_scripts <- function(){
  current_wd <- getwd()
  if (!grepl(PROJECT_FOLDER, getwd())){
    stop(paste("Please set the working directory to a location in the repository", PROJECT_FOLDER))
  } else {
    while (!endsWith(getwd(), PROJECT_FOLDER)){
      setwd("..")
    }
  }
  setwd("./scripts")
  return(current_wd)
}
